other/two_dimensional_materials/layer_on_a_surface.py

```diff
@@ -0,0 +1,144 @@
+from ase.build import surface, supercells
+from ase.io import read, write
+import io
+import numpy as np
+  
+def poscar_to_atoms(poscar):
+    input = io.StringIO(poscar)
+    atoms = read(input, format="vasp")
+    return atoms
+
+
+def write_atoms_to_poscar(atoms):
+    output = io.StringIO()
+    write(output, atoms, format="vasp")
+    content = output.getvalue()
+    output.close()
+    return content
+
+
+def expand_matrix_2x2_to_3x3(matrix_2x2):
+    matrix_3x3 = [[0, 0, 0], [0, 0, 0], [0, 0, 1]]
+
+    for i in range(2):
+        for j in range(2):
+            matrix_3x3[i][j] = matrix_2x2[i][j]
+
+    return matrix_3x3
```

Comment:
Can replace with numpy one-liner here, (np.pad)

Reply:
Yes. Also np.vstack() exists. And I've been using it after

Reply:
Kevin's suggestion seems reasonable:
```py
In [2]: np.pad(np.matrix("1 2; 3 4"), pad_width=((0, 1), (0, 1)))
Out[2]:
array([[1, 2, 0],
       [3, 4, 0],
       [0, 0, 0]])
```

Also, I noticed the same repeating code in another module below. Can those functions be moved to some importable modules?

Reply:
But we also need very last number to be 1 -- no transformations for vector c. So there will be at least another line of adding last vector (0,0,1). 
As for code repetition -- this PR is still WIP so I didn't expect people to look at it closely -- I've been hacking and figuring out stuff in ipynb files with plan to move reusable definitions into py modules. This is still underway